x/auth/keeper/grpc_query_test.go

```diff
@@ -192,73 +193,116 @@ func (suite *KeeperTestSuite) TestGRPCQueryParameters() {
-					var account types.AccountI
-					err := suite.app.InterfaceRegistry().UnpackAny(acc, &account)
-					suite.Require().NoError(err)
-
-					moduleAccount, ok := account.(types.ModuleAccountI)
-
-					suite.Require().True(ok)
-					if moduleAccount.GetName() == "mint" {
-						mintModuleExists = true
-					}
-				}
-				suite.Require().True(mintModuleExists)
+			func(res *types.AddressBytesToStringResponse) {
+				text, err := suite.app.AccountKeeper.GetAddressCdC().BytesToString(addrBytes)
+				suite.Require().NoError(err)
+				suite.Require().NotNil(text)
+				suite.Require().Equal(text, res.AddressString)
 			},
 		},
 		{
-			"invalid module name",
-			func() {
-				req = &types.QueryModuleAccountsRequest{}
-			},
-			true,
-			func(res *types.QueryModuleAccountsResponse) {
-				var mintModuleExists = false
-				for _, acc := range res.Accounts {
-					var account types.AccountI
-					err := suite.app.InterfaceRegistry().UnpackAny(acc, &account)
-					suite.Require().NoError(err)
+			"request is empty",
+			&types.AddressBytesToStringRequest{},
+			false,
+			func(res *types.AddressBytesToStringResponse) {},
+		},
+		{
+			"empty account address in request",
+			&types.AddressBytesToStringRequest{AddressBytes: []byte{}},
+			false,
+			func(res *types.AddressBytesToStringResponse) {},
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			suite.SetupTest() // reset
 
-					moduleAccount, ok := account.(types.ModuleAccountI)
+			ctx := sdk.WrapSDKContext(suite.ctx)
 
-					suite.Require().True(ok)
-					if moduleAccount.GetName() == "falseCase" {
-						mintModuleExists = true
-					}
-				}
-				suite.Require().False(mintModuleExists)
+			res, err := suite.queryClient.AddressBytesToString(ctx, tc.req)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+				suite.Require().NotNil(res)
+			} else {
+				suite.Require().Error(err)
+				suite.Require().Nil(res)
+			}
+
+			tc.posttests(res)
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestAddressStringToBytes() {
+	_, _, addr := testdata.KeyTestPubAddr()
+	addrBytes := []byte(addr)
+	text, err := suite.app.AccountKeeper.GetAddressCdC().BytesToString(addrBytes)
+	suite.Require().NoError(err)
+
+	testCases := []struct {
+		msg       string
+		req       *types.AddressStringToBytesRequest
+		expPass   bool
+		posttests func(res *types.AddressStringToBytesResponse)
+	}{
+		{
+			"success",
+			&types.AddressStringToBytesRequest{AddressString: text},
+			true,
+			func(res *types.AddressStringToBytesResponse) {
+				suite.Require().True(bytes.Equal(res.AddressBytes, addrBytes))
 			},
 		},
+		{
+			"request is empty",
+			&types.AddressStringToBytesRequest{},
+			false,
+			func(res *types.AddressStringToBytesResponse) {},
+		},
+		{
+			"AddressString field in request is empty",
+			&types.AddressStringToBytesRequest{AddressString: ""},
+			false,
+			func(res *types.AddressStringToBytesResponse) {},
+		},
```

Comment:
let's add a test case where the AddressString has a different bech32prefix, e.g. `regen1...`, and the test should fail.

Reply:
Done!